packages/backend/src/core/collectors/PerpetualCairoOutputCollector.ts

```diff
@@ -13,16 +14,31 @@ const ABI = new utils.Interface([
 ])
 
 export class PerpetualCairoOutputCollector {
-  constructor(private readonly ethereumClient: EthereumClient) {}
+  constructor(
+    private readonly ethereumClient: EthereumClient,
+    private readonly starkexConfig: StarkexConfig
+  ) {}
 
-  async collect(transactionHash: Hash256): Promise<PerpetualCairoOutput> {
+  async collect(
+    transactionHash: Hash256,
+    blockNumber: number
+  ): Promise<PerpetualCairoOutput> {
     const tx = await this.ethereumClient.getTransaction(transactionHash)
     if (!tx) {
       throw new Error('Invalid transaction')
     }
     const decoded = ABI.decodeFunctionData('updateState', tx.data)
     const programOutputValues = decoded.programOutput as BigNumber[]
     const hexData = toHexData(programOutputValues)
-    return decodePerpetualCairoOutput(hexData)
+
+    // This is a fix for Apex Goerli testnet, which switches
+    // to a different program output (one more field) at block 8056029.
+    // See: https://github.com/starkware-libs/stark-perpetual/blob/eaa284683deec190407fece98b96546d10f6ad67/src/services/perpetual/cairo/output/program_output.cairo#L36
+    const skipDataAvailabilityModeField =
```

Comment:
I would extract it to separate method called `shouldSkipDataAvailabilityModeField` so it does not make mess in collect function. The comment could be moved there also.

Reply:
Done